File: app/src/main/java/mobsoft/bme/hu/irodafoglalo/ui/UIModule.java
package mobsoft.bme.hu.irodafoglalo.ui;

import android.content.Context;

import java.util.concurrent.Executor;
import java.util.concurrent.Executors;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mobsoft.bme.hu.irodafoglalo.di.Network;
import mobsoft.bme.hu.irodafoglalo.ui.main.MainPresenter;
import mobsoft.bme.hu.irodafoglalo.ui.rooms.RoomsPresenter;

/**
 * Created by mobsoft on 2016. 04. 18..
 */
@Module
public class UIModule {
    private Context context;

    public UIModule(Context context) {
        this.context = context;
    }

    @Provides
    public Context provideContext() {
        return context;
    }

    @Provides
    @Singleton
    public MainPresenter provideMainPresenter() {
        return new MainPresenter();
    }

    @Provides
    @Singleton
    public RoomsPresenter provideArtistsPresenter() {
        return new RoomsPresenter();
    }

    @Provides
    @Singleton
    @Network
    public Executor provideNetworkExecutor() {
        return Executors.newFixedThreadPool(1);
    }
}

File: app/src/main/kotlin/mobsoft/bme/hu/irodafoglalo/ui/UIModule.kt
package mobsoft.bme.hu.irodafoglalo.ui

import android.content.Context
import dagger.Module
import dagger.Provides
import mobsoft.bme.hu.irodafoglalo.ui.main.MainPresenter
import mobsoft.bme.hu.irodafoglalo.ui.participants.ParticipantsPresenter
import mobsoft.bme.hu.irodafoglalo.ui.rooms.RoomsPresenter
import mobsoft.bme.hu.irodafoglalo.ui.times.TimesPresenter
import javax.inject.Singleton

/**
 * Created by Adam Toth on 2016. 04. 19..
 */
@Module
class UIModule(val context: Context) {

    @Provides
    fun provideContext(): Context {
        return context
    }

    @Provides
    @Singleton
    fun provideMainPresenter(): MainPresenter {
        return MainPresenter()
    }

    @Provides
    @Singleton
    fun provideRoomsPresenter(): RoomsPresenter {
        return RoomsPresenter()
    }

    @Provides
    @Singleton
    fun provideTimesPresenter(): TimesPresenter {
        return TimesPresenter()
    }

    @Provides
    @Singleton
    fun provideParticipantsPresenter(): ParticipantsPresenter {
        return ParticipantsPresenter()
    }
}
